### DISCUSSION TITLE
A second run of Prisma migrate (with no schema changes in the interim) reports diff and throws "duplicate key" error

### DISCUSSION THREAD

**Discussion by feather-michael** [prisma/prisma#10389] [Q&A] [answered]
I'm not sure if this should be filed as a bug but it seems very similar to this (how Prisma handles MySql indices): https://github.com/prisma/migrate/issues/148#issuecomment-537106773

Database: MySql
Prisma version: "prisma": "^3.5.0"

Expected outcome: After introspecting the DB and running an initial migration, making no changes of any kind to the schema after that initial introspection, I expect a second run of `npx prisma migrate dev` not to output any second migration file, i.e. it should simply log "Already in sync, no schema change or pending migration was found" (because no schema changes have occurred).

Actual outcome: See below; a second migration file is generated, creating an index which was already created in the first migration, and the migration fails and cannot be applied.

Steps to reproduce:
- Introspect a moderately large (~100 tables), already-existing MySql database: `npx prisma db pull`
- Run the first migration: `npx prisma migrate dev --init`
- (After making no changes of any kind to `prisma.schema`, did the following): Run a second migration immediately after the first `npx prisma migrate dev --second`
- A new migration file reporting a diff is (incorrectly) generated, attempting to create an index which was already part of the first migration
- The second migration obviously cannot be applied and fails:
```
- Database error code: 1061

Database error:
Duplicate key name 'IDX_fileName'
```

What's odd is that there is nothing special or noteworthy about this particular index: it's just one of many indices in this DB, and that index name `IDX_fileName` is unique and exists nowhere else in the database.  The syntax in `schema.prisma` uses exactly the same `@@index` syntax for this index which is used to create all other indices.  The index is not a foreign key but just a regular string column called "fileName":
```
fileName        String         @db.VarChar(1023)
```

To test this I rolled back the failed migration, deleted the entire migrations folder, removed the failing index from `schema.prisma` and attempted the same sequence above (`npx prisma migrate dev`, make no changes to schema, then run `npx prisma migrate dev` again).

On this second test I see the expected outcome: "Already in sync, no schema change or pending migration was found."  No second migration is generated at all because nothing has changed in the schema.

So I have no idea why this one index (out of hundreds in the DB) causes this behavior, but am curious if it has something to do with the bug fix linked at the top which mentions oddities surrounding MySql indices.  

Any ideas/suggestions?

**Comment by austincrim** [comment, marked as answer]
Hey @feather-michael 👋,

This seems unexpected! I am going to convert this discussion into an issue where our engineering teams will pick it up. 

The comment you linked to above definitely feels relevant.

Thanks for bringing this to our attention!

**Reply by feather-michael** [reply]
Thanks @austincrim , I believe this has already been picked up here: https://github.com/prisma/prisma/issues/10405

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
